sourcecode/chesssimul/board.py

```diff
@@ -108,18 +108,18 @@ def move(self, lam: str):
         start_square.piece = None
 
     def make_move(self, move: Move) -> bool:
-        self.move_list.append()
-
-        if move.end[0] in range(0, 8) and move.end[1] in range(0, 8):
-            # postion de la pièce avant son déplacement dans l'échiquier
+        self.move_list.append(self)
+        
+        if (move.end[0] >= 0 and move.end[1] <= 7):
```

Comment:
should be : if move.end[0] >= 0 and move.end[0] <= 7 and move.end[1] >= 0 and move.end[1] <= 7